### DISCUSSION TITLE
Making the <article> element take up more space when both sidebars are collapsed/hidden

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#8794] [Q&A] [answered]
I'm working on making the <article> element take up more space when the left sidebar is collapsed and the right sidebar (TOC) is hidden via the `hide_table_of_contents` frontmatter property.

![image](https://user-images.githubusercontent.com/47085752/226366568-638268a1-9089-45af-8af3-01aec278ab27.png)

Is there a way for me to work with CSS and have the article element expand whenever the TOC sidebar is hidden and the left sidebar is collapsed? Basically, I want <article> to grow when more space is available.

**Comment by slorber** [comment, marked as answer]
The theming system is flexible to allow you to provide your own React and CSS components. I can't really help on implementing custom layouts on a case by case basis: you have to swizzle, and adapt the DOM structure and CSS to your need.

**Reply by ZenaMel** [reply]
I can work with that. Thanks. You don't know what component handles the layout between the sidebars and the main content by chance?

**Reply by slorber** [reply]
DocRoot/Layout contains how we layout sidebar + main content 

(main content does not include TOC because main content is not necessarily a doc, it can be a category index for example)

**Reply by ZenaMel** [reply]
fyi: it seems to have changed from `DocRoot/Layout` -> `DocPage/Layout`
